### ISSUE TITLE
[intl-locales-supported] Intl is not defined

### ISSUE THREAD

**Issue by adrienharnay** [formatjs/formatjs-old#108]
**Which package?**
intl-locales-supported

**Describe the bug**
Since this code is at the root of the file:

```
const DEFAULT_CONSTRUCTORS = [
  Intl.DateTimeFormat,
  Intl.NumberFormat,
  Intl.PluralRules
];
```

It is executed when the module is imported. The code in the README is therefore incorrect:
```
var areIntlLocalesSupported = require('intl-locales-supported');

var localesMyAppSupports = [];

if (global.Intl) {
  if (!areIntlLocalesSupported(localesMyAppSupports)) {
```

It will crash on the require, and should instead be:
```
var localesMyAppSupports = [];

if (global.Intl) {
  var areIntlLocalesSupported = require('intl-locales-supported');

  if (!areIntlLocalesSupported(localesMyAppSupports)) {
```

But I think `Intl` should instead not be accessed from outside the `areIntlLocalesSupported` function.

**To Reproduce**

1. Create a JS project
2. `global.Intl = null`
3. `var areIntlLocalesSupported = require('intl-locales-supported');`

**Expected behavior**
Should not crash when imported

Thanks in advance!

**Comment by longlho** [comment]
Thanks for reporting. Since we only support modern browsers, Node LTS & IE11+ where `Intl` are all available, `intl-locales-supported` primarily checks for **full ICU** + ES2018 `Intl` instead (`Intl.RelativeTimeFormat`). I'll update the docs to reflect this.

**Comment by adrienharnay** [comment]
Oh OK I see, thank you 👍

**Comment by longlho** [comment]
fixed in 5c7daabfe8e29cb6bea7f12243f17bd7796fd295
